plugins/modules/packaging/os/yum_versionlock.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Florian Paul Hoberg <florian.hoberg@credativ.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: yum_versionlock
+version_added: 2.0.0
+short_description: Locks / Unlocks an installed package(s) from being updated by yum package manager.
+description:
+     - This module adds installed packages to yum versionlock to prevent the package from being updated.
+options:
+  name:
+    description:
+      - Package name or a list of packages.
+    type: list
+    required: true
+    elements: str
+  state:
+    description:
+    - If state is C(present) or C(locked), package(s) will be added to yum versionlock list.
+    - If state is C(absent) or C(unlocked), package(s) will be removed from yum versionlock list.
+    choices: [ 'absent', 'locked', 'present', 'unlocked' ]
+    type: str
+    default: present
+notes:
+    - Requires yum-plugin-versionlock package on the remote node.
+    - Supports C(check_mode).
+requirements:
+- yum
+- yum-versionlock
+author:
+    - Florian Paul Hoberg (@florianpaulhoberg)
+'''
+
+EXAMPLES = r'''
+- name: Prevent Apache / httpd from being updated
+  community.general.yum_versionlock:
+    state: present
+    name: httpd
+
+- name: Prevent multiple packages from being updated
+  community.general.yum_versionlock:
+    state: present
+    name:
+    - httpd
+    - nginx
+    - haproxy
+    - curl
+
+- name: Unlock Apache / httpd to be updated again
+  community.general.yum_versionlock:
+    state: absent
+    package: httpd
+'''
+
+RETURN = r'''
+packages:
+    description: A list of package(s) in versionlock list.
+    returned: always
```

Comment:
This return value is not always returned. Neither the `fail_json` call in `module.get_bin_path` nor the explicit `fail_json` calls in this module return it.

Reply:
Realized that now, fixed in https://github.com/ansible-collections/community.general/pull/1405/commits/39db7f619cfc2dddae18d80a1ce4ac16c49fcf6f.